File: src/Lavcode.Service.Api/Dtos/UpdateFolderDto.cs
using Newtonsoft.Json;

namespace Lavcode.Service.Api.Dtos
{
    public class UpdateFolderDto
    {
        [JsonProperty("name")]
        public string Name { get; set; }

        [JsonProperty("order")]
        public int Order { get; set; }

        [JsonProperty("icon")]
        public UpsertIconDto Icon { get; set; }
    }
}

File: src/lavcode-node/src/actions/folder/dtos/update-folder.dto.ts
import { V } from "@halsp/validator";
import { UpsertIconDto } from "../../../dtos/upsert-icon.dto";

export class UpdateFolderDto {
  @V().Required().IsString()
  name!: string;

  @V().Required().IsNumber()
  order!: number;

  @V().IsOptional().Description("空则不修改图标")
  icon!: UpsertIconDto;
}
